XenAdmin/Wizards/PatchingWizard/PatchingWizard_PrecheckPage.cs

```diff
@@ -954,5 +973,32 @@ private void dataGridView1_CellMouseMove(object sender, DataGridViewCellMouseEve
             else
                 Cursor = Cursors.Arrow;
         }
+
+        private void refreshDataGridViewList()
+        {
+            if(!checkBoxViewPrecheckFailuresOnly.Checked)
+            {
+                foreach (var row in succeessfulCheck)
+                {
+                    if (row.Key < dataGridView1.Rows.Count && dataGridView1.Rows[row.Key] != row.Value)
+                    {
+                        InsertRowToGridView(row.Key, row.Value);
+                    }
+                    else if (row.Key == dataGridView1.Rows.Count)
+                    {
+                        AddRowToGridView(row.Value);
+                    }
+                }
+            }
+            else
+            {
+                foreach (var row in succeessfulCheck.Reverse())

```

Comment:
I quite don't understand .Reverse() here. successfulCheck is a Dictionary, not a List. Is your intention to reverse a list here?

Reply:
This is because the implementation of Dictionary of C#. If successfulCheck has 3 items, and you use `<                foreach (var row in successfulCheck)
                {
                        dataGridView1.Rows.Remove(row.Value);
                }
>` It will delete the item 0 and item2, the item 1 will be left.